### ISSUE TITLE
[RFC] File namer for uniquely suffixing a file name if it is already taken

### ISSUE THREAD

**Issue by benglass** [dustin10/VichUploaderBundle#417]
A common piece of functionality for file upload would be to keep the file's original name unless there is already a file with that name, in which case add a unique suffix (numeric or something like that). This would be similar to how browsers handle file downloads when you already have a file of that name. I have a namer that already does this if there is interest I can clean it up and submit a PR.

**Comment by K-Phoen** [comment]
Yup, I also think that this feature would be useful. Feel free to submit a PR :+1:

**Comment by K-Phoen** [comment]
Ping ?

**Comment by benglass** [comment]
Let me see if I can whip up a PR this weekend

**Comment by K-Phoen** [comment]
Perfect :)

Don't hesitate to open a PR even if it's still WIP, I can help you if you don't have enough time.

**Comment by garak** [comment]
Any news on this?

**Comment by Koc** [comment]
Namers hasn't know anything about storage. We need inject storage into namer for checking if file already exists. But this is not enough: we also need onject directory namer for getting full path. I think that this is hard to resolve and you can just use `HashNamer` from #622
